cmd/edgectl/aes_install.go

```diff
@@ -437,166 +428,97 @@ func (i *Installer) Perform(kcontext string) error {
+		i.Report("fail_download", ScoutMeta{"err", err.Error()})
+		return errors.Errorf("could not download")
+	}
 
-		if err := i.ShowKubectl("install CRDs", crdManifests, "apply", "-f", "-"); err != nil {
-			i.Report("fail_install_crds")
-			return err
-		}
+	// the AES version we have downloaded
+	downVersion := strings.Trim(chartDown.GetChart().AppVersion, "\n")
 
-		if err := i.ShowKubectl("wait for CRDs", "", "wait", "--for", "condition=established", "--timeout=90s", "crd", "-lproduct=aes"); err != nil {
-			i.Report("fail_wait_crds")
-			return err
-		}
+	chartRel, err := chartDown.GetReleaseMgr(defInstallNamespace, chartValues)
+	defer func() { _ = chartDown.Cleanup() }()
+	if err != nil {
+		i.log.Printf("when obtaining the chart manager: %s", err)
+		i.Report("fail_download", ScoutMeta{"err", err.Error()})
+		return err
+	}
 
-		if err := i.ShowKubectl("install AES", aesManifests, "apply", "-f", "-"); err != nil {
-			i.Report("fail_install_aes")
-			return err
-		}
+	if err := chartRel.Sync(i.ctx); err != nil {
+		i.log.Printf("Failed to sync release: %s", err)
+		i.Report("fail_download", ScoutMeta{"err", err.Error()})
+		return err
+	}
+
+	if !chartRel.IsInstalled() {
+		// Ambassador is not installed at all: perform the installation
+		i.SetMetadatum("AES version being installed", "aes_version", downVersion)
+		i.version = downVersion
+		i.ShowWrapped(fmt.Sprintf("-> Installing the Ambassador Edge Stack %s.", i.version))
 
-		if err := i.ShowKubectl("wait for AES", "", "-n", "ambassador", "wait", "--for", "condition=available", "--timeout=90s", "deploy", "-lproduct=aes"); err != nil {
-			i.Report("fail_wait_aes")
+		installedRelease, err := chartRel.InstallRelease(i.ctx)
+		if err != nil {
+			i.log.Printf("Installation of a new release failed: %s", err)
+			i.Report("fail_install_aes", ScoutMeta{"err", err.Error()})
 			return err
 		}
-	}
 
-	// Wait for Ambassador Pod; grab AES install ID
-	if err := i.loopUntil("AES pod startup", i.GrabAESInstallID, lc2); err != nil {
-		i.Report("fail_pod_timeout")
-		return err
-	}
+		i.log.Printf("New release installed successfully")
+		i.log.Printf("Config values: %+v", installedRelease.Config)
 
-	// Grab Helm information if present
-	if managedDeployment, err := i.CaptureKubectl("get deployment labels", "", "get", "-n", "ambassador", "deployments", "ambassador", "-Lapp.kubernetes.io/managed-by"); err == nil {
-		if strings.Contains(managedDeployment, "Helm") {
-			i.SetMetadatum("Cluster Info", "managed", "helm")
+		message := ""
+		if installedRelease.Info != nil {
+			message = installedRelease.Info.Notes
 		}
+		i.ShowWrapped(message, "\n")
+		i.Report("deploy")
+	} else {
+		// Ambassador is currently installed. Don't do anything.
+		i.ShowWrapped(fmt.Sprintf("-> Found an existing installation of Ambassador Edge Stack %s.", downVersion))
+		i.ShowWrapped("-> ... will not try to install/update Ambassador")
+		i.Report("installed")
```

Comment:
This is a new report. The old code reported either `deploy` or a failure. We should be careful about changing that. Whatever we do, `metq` will have to support the old and new reports for a long time.

Reply:
I can see the confusion between `deploy` and `installed`. Maybe it would make it clearer to rename this one to `already_installed` or something that will better represent this new code path.

Reply:
This isn't really a new code path from the metrics perspective. If we want to distinguish between "installed this run" and "installed prior to this run" I suggest we use a metadata item. In other words, I say we always report `deploy` if a deployment exists (like the old installer) and add `already_installed: true` or `already_installed: false` metadata as is appropriate.

Reply:
Ok, I can add that kind of metrics.

I think it would be convenient to document all these magic values for metrics. Something like:

```go
const (
    // reportDeployed is sent when the deployment exists
    reportDeployed = "deploy"

    // reportAlreadyInstalled is sent when a deployment was already installed in the system
    reportAlreadyInstalled = "already_installed"
)
```